packages/query/src/graphql/plugins/PgDistinctPlugin.ts

```diff
@@ -88,6 +89,10 @@ export const PgDistinctPlugin: Plugin = (builder) => {
 
             // Dependent on https://github.com/graphile/graphile-engine/pull/805
             (queryBuilder as any).distinctOn(id);
+
+            if (argv('dictionary-optimisation')) {
+              queryBuilder.setOrderIsUnique();
```

Comment:
Can you please add a comment as to why we do this.

Reply:
Is it always good to do this, not just in case of dictionary?

Reply:
what if there is a case where user wants to use `distinct` but orderedBy primaryKey